src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilationWithAnalyzers.cs

```diff
@@ -711,48 +703,29 @@ private async Task ComputeAnalyzerDiagnosticsAsync(AnalysisScope? analysisScope,
             {
                 throw ExceptionUtilities.Unreachable();
             }
-            finally
-            {
-                if (driverAllocatedFromPool)
-                    FreeDriver(driver);
-            }
 
-            async Task<(Compilation compilation, AnalyzerDriver driver, bool driverAllocatedFromPool)> getCompilationAndDriverAsync()
+            async Task<(Compilation compilation, AnalyzerDriver driver)> getCompilationAndDriverAsync()
             {
                 Debug.Assert(analysisScope != null);
 
-                Compilation compilation;
-                AnalyzerDriver driver;
-                bool driverAllocatedFromPool;
-
-                // We reuse the root "_compilation" when performing purely syntactic analysis or computing semantic diagnostics
-                // only for the compiler analyzer.
+                // We reuse the root "_compilation" when computing diagnostics only for the compiler analyzer.
                 // For rest of the cases, we use a cloned compilation with a new event queue to allow us to execute analyzers on this
                 // compilation without any partial state tracking and subsequently discard this compilation.
-                if (analysisScope.IsSyntacticSingleFileAnalysis ||
-                    analysisScope.IsSemanticSingleFileAnalysisForCompilerAnalyzer)
+                Compilation compilation;
+                if (analysisScope.IsSingleFileAnalysisForCompilerAnalyzer)
                 {
                     compilation = _compilation;
-                    driver = await GetAnalyzerDriverFromPoolAsync(cancellationToken).ConfigureAwait(false);
-                    driverAllocatedFromPool = true;
                 }
                 else
                 {
                     // Clone the compilation with new event queue and semantic model provider.
                     compilation = _compilation.WithSemanticModelProvider(new CachingSemanticModelProvider()).WithEventQueue(new AsyncQueue<CompilationEvent>());
-
-                    // Get the analyzer driver to execute analysis.
-                    driver = CreateDriverForComputingDiagnosticsWithoutStateTracking(compilation, originalAnalyzers);
-                    driverAllocatedFromPool = false;
-
-                    // Initialize the driver.
-                    driver.Initialize(compilation, _analysisOptions, new CompilationData(compilation), categorizeDiagnostics: true, trackSuppressedDiagnosticIds: false, cancellationToken);
-                    cancellationToken.ThrowIfCancellationRequested();
-
-                    await driver.WhenInitializedTask.ConfigureAwait(false);
                 }
```

Comment:
nit: could be a `? :` expression.

Reply:
Going to keep it as-is to retain the comment in the else part.